lib/charms/vault_k8s/v0/juju_facade.py

```diff
@@ -502,11 +591,16 @@ def set_app_relation_data(
             data=data,
             relation_name=relation_name,
             relation_id=relation_id,
+            relation=relation,
             entity=self.charm.model.app,
         )
 
     def set_unit_relation_data(
-        self, data: dict[str, str], relation_name: str, relation_id: int | None
+        self,
+        data: dict[str, str],
+        relation_name: str,
+        relation_id: int | None = None,
```

Comment:
The `relation` prefixes are unnecessary, it's obvious we are talking about the relation. 

The same may apply elsewhere

Reply:
Fixed